src/protocol/boolean/random_bits_generator.rs

```diff
@@ -54,12 +90,15 @@ where
     /// inner members multiple times, I/O errors while executing MPC protocols,
     /// read from an empty buffer, etc.
     pub async fn generate(&self) -> Result<RandomBitsShare<F, S>, Error> {
```

Comment:
At the call site for this function, I think that we should close out the fallback generator.

Reply:
given that we know how many records this generator is expected to process, we could close it inside `generate` once the last record is processed

would it be easier to work with? I've pushed this version, please take a look and let me know what you think